### PR TITLE
1990S - Form updates

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
- fixed several labels
- updated to newest direct deposit UI
- fix typo

## Screenshots
Screen Shot 2021-04-01 at 2 48 19 PM
Screen Shot 2021-04-01 at 2 48 16 PM
Screen Shot 2021-04-01 at 2 48 11 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by U-DON — APPROVED** [review]

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx [hunk 1]
```diff
@@ -38,12 +38,6 @@ const uiSchema = ({ affectedBenefits, unaffectedBenefits, optionalFields }) => {
         </div>
       ) : null;
     },
-    'ui:order': [
-      'bankAccount',
-      'declineDirectDeposit',
-      'view:directDespositInfo',
-      'view:bankInfoHelpText',
-    ],
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/definitions/directDeposit.jsx:46]: Why is this being removed?

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx [hunk 2]
```diff
@@ -114,23 +100,11 @@ const uiSchema = ({ affectedBenefits, unaffectedBenefits, optionalFields }) => {
     },
   };
 
-  if (!optionalFields.declineDirectDeposit) {
-    // We're not using declineDirectDeposit; Remove the entry from ui:order so
-    // it doesn't error out
-    const i = ui['ui:order'].indexOf('declineDirectDeposit');
-    if (i !== -1) ui['ui:order'].splice(i, 1);
-  }
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/definitions/directDeposit.jsx:122]: Why is this being removed?

**Review by bkjohnson — COMMENTED** [review]
There are some more changes being made to the directDeposit definition that I didn't leave comments on. Can you explain why you're making them?

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx (hunk 1 above)
Comment (zurbergram) [reply src/platform/forms-system/src/js/definitions/directDeposit.jsx:46]: image
on the review page of 1990s i was receiving this error message regarding the view properties in both `ui:order`

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx (hunk 2 above)
Comment (zurbergram) [reply src/platform/forms-system/src/js/definitions/directDeposit.jsx:122]: Related to removing `ui:order` in general

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx (hunk 1 above)
Comment (zurbergram) [reply src/platform/forms-system/src/js/definitions/directDeposit.jsx:46]: Change was done to schema creation to ensure order is correct

Comment (zurbergram) [reply src/platform/forms-system/src/js/definitions/directDeposit.jsx:46]: I attempted to remove the properties from `ui:order` which allowed for the Review page to render but did allow for the page in the form to render.

I tried using as well
```
'ui:options': {
        hideOnReview: true,
      },
```

**Review by delli-gatti-michael — APPROVED** [review]

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx (hunk 1 above)
Comment (bkjohnson) [reply src/platform/forms-system/src/js/definitions/directDeposit.jsx:46]: Based on that screenshot it looks like you only needed to remove the properties starting with `view:`.

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx [hunk 3]
```diff
@@ -139,6 +113,11 @@ const uiSchema = ({ affectedBenefits, unaffectedBenefits, optionalFields }) => {
   return ui;
 };
 
+/**
+ * Constructs schema in "sections" to perserve order since the Review page doesn't like view properties
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/definitions/directDeposit.jsx:117]: What does "the Review page doesn't like view properties" mean? We have `view` properties in other forms that don't seem to be causing any problems on the review page.

File: src/platform/forms-system/src/js/definitions/directDeposit.jsx [hunk 4]
```diff
@@ -175,13 +145,34 @@ const schema = optionalFields => {
       ? optionalFields.declineDirectDeposit.schema
       : { type: 'boolean' };
   }
+
+  s.properties = {
+    ...s.properties,
+    'view:directDepositInfo': { type: 'object', properties: {} },
+    'view:bankInfoHelpText': { type: 'object', properties: {} },
+  };
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/definitions/directDeposit.jsx:153]: I don't see how this is any different from including the `view` properties up above in the original declaration for `s`.

Comment (zurbergram) [reply src/platform/forms-system/src/js/definitions/directDeposit.jsx:153]: `directDepositUiSchema` comes from https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/definitions/directDeposit.jsx#L41
 
If I remove or null out `'ui:order'`s 

Screen Shot 2021-04-05 at 3 57 53 PM

then the `declineDirectDeposit` is not in the correct location.

Screen Shot 2021-04-05 at 3 57 44 PM


The reason for this is that without an `'ui:order'` property the objects within `uiSchema` are rendered in the order in which the properties are added to the `schema`

**Comment by zurbergram** [comment]
After filling out the form and proceeding to the review and submit page, upon expanding the accordion to review the page with direct deposit information,

This is what happens 
Screen Shot 2021-04-05 at 4 09 24 PM

the error messages state

```
Uncaught Error: uiSchema order list contains extraneous properties 'view:directDepositInfo', 'view:bankInfoHelpText'
    at orderProperties (utils.js?8964:306)
    at ObjectField.eval (ObjectField.jsx?2f0d:36)
    at ObjectField.eval [as orderAndFilterProperties] (_createFlow.js?3414:68)
    at ObjectField.render (ObjectField.jsx?2f0d:102)
    at finishClassComponent (react-dom.development.js?61bb:17160)
    at updateClassComponent (react-dom.development.js?61bb:17110)
    at beginWork (react-dom.development.js?61bb:18620)
    at HTMLUnknownElement.callCallback (react-dom.development.js?61bb:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:237)
    at invokeGuardedCallback (react-dom.development.js?61bb:292)
```

If i remove the view properties from the `'ui:order'` then I get the same error but on the form page
Screen Shot 2021-04-05 at 4 11 39 PM

**Comment by zurbergram** [comment]
The error on the review page is coming from https://github.com/department-of-veterans-affairs/react-jsonschema-form/blob/master/src/utils.js#L234

**Comment by zurbergram** [comment]
It looks like the view fields are not in [propertyHash](https://github.com/department-of-veterans-affairs/react-jsonschema-form/blob/master/src/utils.js#L232)

Some how the view fields are being filtered out of the schema when it is being passed into https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/review/ObjectField.jsx#L75

**Comment by zurbergram** [comment]
Removed majority of changes from `src/platform/forms-system/src/js/definitions/directDeposit.jsx`.  Nulling out the `'ui:order'` and redeclaring schema in correct order

**Pull request merged**